### DISCUSSION TITLE
How do you use the scalene extension with VS Code?

### DISCUSSION THREAD

**Discussion by jmfontana** [plasma-umass/scalene#703] [Q&A] [answered]
When I introduce the lines:

scalene_profiler.start()
and
scalene_profiler.stop()

at the beginning and end of my python scripts and run them from VS Code, I get this error:

"ERROR: Do not try to invoke `start` if you have not called Scalene using one of the methods
in https://github.com/plasma-umass/scalene#using-scalene
(The most likely issue is that you need to run your code with `scalene`, not `python`)."

That doesn't happen if I run it on the terminal by doing:

% scalene script.py

but then, I wonder, how does one profile scripts with the VS Code extension?

**Comment by emeryberger** [comment]
To use the extension, you don't modify your code at all! You just run the extension on your program.

**Comment by jmfontana** [comment]
Thanks. I'm not sure what you mean by running the extension on my program. I have the scalene extension installed in VS Code but when I execute a script, I don't get any 'profile.html' created. I only get this file to be created and the browser open it automatically if I run scalene from the terminal.

**Comment by emeryberger** [comment, marked as answer]
You need to click Command-Shift-P (or Ctrl-Shift-P) to open the [Command Palette](https://code.visualstudio.com/docs/getstarted/userinterface). Then select "Scalene: AI-powered profiling..." (you can start typing Scalene and it will pop up if it's installed). 

<img width="734" alt="Screenshot 2023-09-20 at 7 09 06 PM" src="https://github.com/plasma-umass/scalene/assets/1612723/7e78e3d2-e649-4f02-86fd-0da2a259a1a4">

Run that and (assuming your code runs for a second or more) you will get a profile in a webview.

<img width="1071" alt="Screenshot 2023-09-20 at 7 32 11 PM" src="https://github.com/plasma-umass/scalene/assets/1612723/79d7e32f-7758-4920-a144-99bb2bfb152c">

**Comment by jmfontana** [comment]
Thanks,

It should work as you say but in my case it seems something is amiss. When I try, I get this error message:

[Process exited with error code](https://ibb.co/0ns2BdL)

JM

----
Moltes gràcies per l'ajut i per la bona feina, Emery. Aquest profiler és excel·lent. Acabo d'escoltar el podcast Real Python i veig que et podria haver fet la pregunta en català.

**Reply by emeryberger** [reply]
Graciès, JM! He vist aquest mateix error abans que he pogut resoldre simplement per instal.lant la ultima versió de Scalene (i, potser, si no recordo mal, en canviant la selecció de l'interpreter en l'extensió Python).

---
Thanks, JM! I've seen this error before, which I was able to solve by installing the latest version of Scalene, and, if I remember correctly, changing the selected Python interpreter used by the Python extension.

**Reply by FernandaRobles1234** [reply]
Hello! I'm having the same problem :)

But I'm working with the latest version (V0.0.9) whish supports python 3.8|3.9|3.10|3.11|3.12 and I'm using the Microsoft store's python 3.12.7. It still doesn't work.

I was wondering if you would have some advice for me.

**Reply by emeryberger** [reply]
I spent some time chasing down the issue and I believe fixed it (I am getting working profiles in the web view). You need to update to the latest version of both Scalene (1.5.47) and the Scalene VS Code extension (0.0.10).

**Reply by jmfontana** [reply]
Thanks Emery.  I've updated the extension and the version of Scalene but I still seem to be having problems.  It's been a while since I used Scalene so there might be something I'm not remembering correctly. Is the profile in the Webview supposed to pop up by itself once the script is finished running? I'm not getting any error messages but I'm not seeing any webview, either, even though I select "Scalene: AI-powered profiling..." before running the script.
